### ISSUE TITLE
The combobox is broken

### ISSUE THREAD

**Issue by AttilaZsamboki** [shadcn-ui/ui#1754]
I don't know how this didn't got noticed  before but the combobox is basically useless. You can't search for anything else if you search for something that doesn't exists. Even on the home page its bugged...

https://github.com/shadcn-ui/ui/assets/99214950/4b59c970-5f97-4e68-a0ef-b4bc9613b1ab

**Comment by kbumsik** [comment]
It is just dummy UI. I guess you should implement your own proper search functionality.

**Comment by AttilaZsamboki** [comment]
Obviously.... the issue is that the combobox is bugged and if i implement it in my own ui it will still be bugged.

**Comment by ItzaMi** [comment]
If you're worried about functionality, refer to https://ui.shadcn.com/docs/components/combobox, please. If there's anything that you might copy, is that one, which works pretty well

**Comment by AttilaZsamboki** [comment]
No, it doesn't work either. I don't mean to sound arrogant, because, after all, it is a free and open-source tool. However, it is buggy. Just watch this clip. I can't search for "next," even though Next.js is in the list. This is because, after you search for something that doesn't exist, you won't be able to search for things that do either.

**Comment by Riculum** [comment]
I am experiencing a similar problem. It occurs when there is a typo or I want to remove part of my search term, but no alternative suggestions are displayed. To see the full list of suggestions, I have to delete my entire input.

**Comment by ItzaMi** [comment]
If you do create the sandbox, and are able to reproduce the issue, you can always try to find the bug and provide a fix for it.
More than anything, the issue is probably with the React logic and not inherent to the UI components

**Comment by ItzaMi** [comment]
So I was experimenting with the issue and the problem seems to be the connection between the ui component and the primitive for `cdmk`.
I recommend you change the existing code to have an extra `value={framework.value}`, so it should look like this:
```javascript
<CommandItem
  key={framework.value}
  onSelect={(currentValue) => {
    setValue(currentValue === value ? "" : currentValue)
    setOpen(false)
  }}
  value={framework.value}
>
```

**Comment by miquelvir** [comment]
Most probably would be fixed by #1450
